### ISSUE TITLE
Issue on outputting key presses on Mac

### ISSUE THREAD

**Issue by fell-lucas** [kmonad/kmonad#687]
Hello, so today I met KMonad and just got to know about it and how awesome it is! 

I read a lot on the past few hours on how to compile on a mac. I managed to do it successfully, the binary is working great, registering keypresses and all, I can even define a specific input device, instead of all of them.

However, when I run KMonad with debug on as you normally would, `sudo kmonad keymap/template/apple.kbd -l debug` I can see the program receiving the keypresses on the terminal:

```
sudo kmonad keymap/template/apple.kbd -l debug
Password:
Opening KeySink
Initializing Mac key sink
Opening KeySource
Opening IOKit devices
Launching process: emitter_proc
connected

--------------------------------------------------------------------------------
Received event: Press <a>
Running hooks
Registering untimed hook: 1
Emitting: Press <a>

--------------------------------------------------------------------------------
Received event: Release <a>
Running hooks
Emitting: Release <a>
qada
--------------------------------------------------------------------------------
Received event: Press <a>
Running hooks
Registering untimed hook: 2
Emitting: Press <a>

--------------------------------------------------------------------------------
Received event: Release <a>
Running hooks
Emitting: Release <a>

--------------------------------------------------------------------------------
Received event: Press <sft>
Running hooks
Registering untimed hook: 3
Emitting: Press <sft>

--------------------------------------------------------------------------------
Received event: Press <i>
Running hooks
Registering untimed hook: 4
Registering untimed hook: 5
Emitting: Press <i>

--------------------------------------------------------------------------------
Received event: Release <sft>
Running hooks
Registering untimed hook: 6
Emitting: Release <sft>

--------------------------------------------------------------------------------
Received event: Release <i>
Running hooks
Emitting: Release <i>

--------------------------------------------------------------------------------
Received event: Press <s>
Running hooks
Registering untimed hook: 7
Emitting: Press <s>

--------------------------------------------------------------------------------
Received event: Release <s>
Running hooks
Emitting: Release <s>

--------------------------------------------------------------------------------
Received event: Press <s>
Running hooks
Registering untimed hook: 8
Emitting: Press <s>

--------------------------------------------------------------------------------
Received event: Release <s>
Running hooks
Emitting: Release <s>
^CClosing process: emitter_proc
Closing KeySource
Closing IOKit devices
Closing KeySink
Closing Mac key sink
Encountered error in KeySource: [Cannot interpret mac switch: 25869298500096,Cannot translate from mac keycode: (28,0)]

```

The error shown in the final lines of the log I presume happened because I `Ctrl+C`'d... It didn't crash or anything while I typed. 

But the problem is, the activity is only happening on kmonad's terminal, in other applications the keyboard is completely unresponsive, as though it is not even plugged in. Not even the caps lock LED is switching on/off when I press it.

Here's a demo video on what is going on: I am pressing keys, but there is no output in the terminal I am focusing... Does anyone know why it is doing that?

https://user-images.githubusercontent.com/47724710/223579173-960681d0-2e55-4079-bc58-03dce7065d3a.mov

**Comment by KaitlynEthylia** [comment]
> However, when I run KMonad with debug on as you normally would, `sudo kmonad keymap/template/apple.kbd -l debug`

I'm unsure if this is the cause of your issue but Kmonad isn't supposed to be run as root, the FAQ provides instructions on how to give uinput permissions so you can run it normally

**Comment by fell-lucas** [comment]
It is on Mac, though: https://github.com/kmonad/kmonad/blob/master/doc/installation.md#giving-kmonad-additional-permissions

**Comment by jaroslaw-weber** [comment]
Have you ever solved this issue? I'm also encountering it. I'm using m1

**Comment by karim-bouchez** [comment]
@fell-lucas @jaroslaw-weber 

I had the same problem (on mac M1), I mostly found the solution on this [thread](https://github.com/kmonad/kmonad/issues/502)


TL;DR: 

deactivate/reactivate `dext`:
```bash
/Applications/.Karabiner-VirtualHIDDevice-Manager.app/Contents/MacOS/Karabiner-VirtualHIDDevice-Manager deactivate
/Applications/.Karabiner-VirtualHIDDevice-Manager.app/Contents/MacOS/Karabiner-VirtualHIDDevice-Manager forceActivate
```

then reboot the system (it's asked by the output of the last command)

**Comment by jaroslaw-weber** [comment]
i tried restarting before and it didnt work. i eventually gave up. i just try to use my programmable keyboard anyway, for small changes i use ukulele mapping
